frontend/src/pages/stakingOverview/AuthzDelegations.js

```diff
@@ -0,0 +1,211 @@
+import React, { useEffect } from "react";
+import Button from "@mui/material/Button";
+import { StyledTableCell, StyledTableRow } from "../../components/CustomTable";
+import Paper from "@mui/material/Paper";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import { formatValidatorStatus } from "../../utils/util";
+import Typography from "@mui/material/Typography";
+import { useTheme } from "@emotion/react";
+import { CircularProgress } from "@mui/material";
+import PropTypes from "prop-types";
+
+export default function AuthzDelegations(props) {
+  const {
+    delegations,
+    validators,
+    currency,
+    undelegateAuthzGrants,
+    redelegateAuthzGrants,
+    granter,
+    rewards,
+    setTotalRewards,
+    onDelegationAction,
+  } = props;
+
+  const theme = useTheme();
+
+  const [rewardsP, setRewardsP] = React.useState({});
+
+  useEffect(() => {
+    let total = 0.0;
+    if (rewards?.length > 0) {
+      for (let i = 0; i < rewards.length; i++) {
+        if (rewards[i].reward.length > 0) {
+          const reward = rewards[i].reward;
+          for (let j = 0; j < reward.length; j++) {
+            if (reward[j].denom === currency.coinMinimalDenom) {
+              let temp = rewardsP;
+              temp[rewards[i].validator_address] =
+                parseFloat(reward[j].amount) / 10 ** currency?.coinDecimals;
+              setRewardsP(temp);
+              total +=
+                parseFloat(reward[j].amount) / 10 ** currency?.coinDecimals;
+            }
+          }
+        } else {
+          let temp = rewardsP;
+          temp[rewards[i].validator_address] = 0.0;
+          setRewardsP(temp);
+        }
+      }
+    }
+
+    setTotalRewards(total.toFixed(5));
+  }, [rewards]);
+
+  const handleClick = (event, type, delegation) => {
+    let val = {};
+    if (delegation.validator_address in validators?.active) {
+      val = validators?.active[delegation.validator_address];
+    } else {
+      val = validators?.inactive[delegation.validator_address];
+    }
+    onDelegationAction(event, type, val);
+  };
+
+  return (
+    <>
+      <TableContainer component={Paper} elevation={0}>
+        {delegations?.status === "pending" ? (
+          delegations?.delegations?.delegations?.length === 0 ? (
```

Comment:
lets assign to some variable and use it from there.
let dels = delegations?.delegations?.delegations;